index.bs

```diff
@@ -0,0 +1,221 @@
+
+A user agent implementing {{LayoutShift}} must perform the following steps:
+1. Run the <a>register a performance entry type</a> algorithm with <code>"layoustShift"</code> as input.
+
+Processing model {#sec-processing-model}
+========================================
+
+Modifications to the HTML specification {#sec-modifications-HTML}
+--------------------------------------------------------
+
+<em>This section will be removed once the <a href=https://html.spec.whatwg.org/multipage>HTML specification</a> has been modified.</em>
+
+For each <a>Element</a>, there is a:
+* <dfn>current frame starting point</dfn> which is initially set to <code>null</code>.
+* <dfn>previous frame starting point</dfn> which is initially set to <code>null</code>.
+* <dfn>current visual representation</dfn> which is initially set to <code>null</code>.
+* <dfn>previous visual representation</dfn> with is initially set to <code>null</code>.
+
+<div algorithm="additions to update rendering">
+    In the <a>update the rendering</a> step of the <a>event loop processing model</a>, add a step right after the step that calls <a>mark paint timing</a>:
+
+    1. For each fully active {{Document}} in <em>docs</em>, invoke the algorithm to <a>evaluate the layout instability</a> of that {{Document}}.
+</div>
+
+Evaluate layo ut instability {#sec-eval-layout-instability}
+--------------------------------------------------------
+
+<div algorithm="evaluate the layout instability">
+    When asked to <dfn export>evaluate the layout instability</dfn> given an active {{Document}} <var>doc</var>, run the following steps:
+
+    1. Let <var>elements</var> be the set of <a>shadow-including descendants</a> of the <var>doc</var>.
+    1. Let <var>unstableElements</var> be an empty list.
+    1. Let <var>totalWidth</var> be the current <a>visual viewport width</a>.
+    1. Let <var>totalHeight</var> be the current <a>visual viewport height</a>. 
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <var>element</var>'s <a>current frame starting point</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <var>element</var>'s <a>current visual representation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>current frame starting point</a> and <a>previous frame starting point</a>. 
+        1. Set <var>element</var>'s <var>shiftFraction</var> to:
+            1. 0 if <a>previous frame starting point</a> is null.
+            1. The max distance the element has moved in any direction computed as the max(abs(<a>previous frame starting point</a>'s y-value - <a>current frame starting point</a>'s y-value) / max(<var>totalWidth</var>, <var>totalHeight</var>), abs(<a>previous frame starting point</a>'s x-value - <a>current frame starting point</a>'s x-value) / max(<var>totalWidth</var>, <var>totalHeight</var>)).
```

Comment:
This looks too complex, it should probably be split into smaller pieces.

Reply:
done.